jazz_metrics/index.js

```diff
@@ -0,0 +1,303 @@
+                    var userStatistics = eventBody.statistics.toLowerCase();
+
+                    // Massaging data from assets api , to get required list of assets which contains type, asset_name and statistics.
+                    var assetsArray = utils.getAssetsObj(apiAssetsArray,userStatistics);
+                    resolve(assetsArray);
+                }
+            }
+        });
+    });
+}
+
+function validateAssets(assetsArray, eventBody) {
+    return new Promise((resolve, reject) => {
+        if(assetsArray.length > 0){
+            var newAssetArray = [];
+            var invalidTypeCount = 0;
+            assetsArray.forEach(function(assetItem){
+                if(assetItem.isError){
+                    logger.error(assetItem.isError);
+                    invalidTypeCount++;
+                    if( invalidTypeCount === assetsArray.length ){
+                        reject({
+                            result: "inputError",
+                            message: "Unsupported metric type."
+                        });
+                    }
+                }
+                else{
+                    // Forming object with parameters required by cloudwatch getMetricStatistics api.
+                    var commonParam = {
+
+                        Namespace: assetItem.type,
+                        MetricName : "",
+                        Period: eventBody.interval,
+                        EndTime: eventBody.end_time,
+                        StartTime: eventBody.start_time,
+                        Dimensions: [{
+                            Name: '',
+                            Value: assetItem.asset_name
+                        }, ],
+                        Statistics: [
+                            assetItem.statistics,
+                        ],
+                        Unit : ""
+                    };
+
+                    var actualParam = [];
+                    var paramMetrics = [];
+                    var missingAssetNameFields;
+                    var intervalPeriod = '';
+                    var assetNameObj = assetItem.asset_name;
+                    var getAssetNameDetails = utils.getNameSpaceAndMetriDimensons(assetItem.type);
+
+                    commonParam.Namespace = getAssetNameDetails.awsNameSpace;
+
+                    if(!getAssetNameDetails.isError){
+                        paramMetrics = getAssetNameDetails.paramMetrics;
+                    }
+                    else{
+                        logger.error("Unsupported metric type. ");
+                        reject({
+                            result: "inputError",
+                            message: "Unsupported metric type."
+                        });
+                    }
+
+                    // Forming the Dimension array from assetNameObj
+                    paramMetrics.forEach(function(arrayItem) {
+                        var clonedObj = {};
+                        for(var key in commonParam){
```

Comment:
try to use let inside blocks(if, for ,etc)

Reply:
done.